### PR TITLE
Image building

### PR DESCRIPTION
**Pull request by warsaw**
This adds an abstraction over the dd and sgdisk commands.  It's fairly low-level at the moment, but we could build higher level abstractions on top of that, or just drive it from image.yml.  That part's not tied in yet.  You should see how `proto.sh` is a pretty straight up translation to Python in `proto.py`.  See the examples directory.

### CODE REVIEW

File: examples/proto.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python3
+
+from ubuntu_image.image import Diagnostics, GiB, Image
+
+# Create empty image of a fixed size
+# TODO: compute rough good size after seeing all the required snaps.
+
+image = Image('img', GiB(4))
```
Comment (zyga) [inline examples/proto.py]: I love the GiB :-)

File: examples/proto.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# FIXME: embed a pointer to 2nd stage in bios-boot partition
+image.copy_blob('blogs/img.mbr', bs=446, count=1, conv='notrunc')
+
+# Create BIOS boot partition
+#
+# The partition is 1MiB in size, as recommended by various partitioning guides.
+# The actual required size is much, much smaller.
+#
+# https://www.gnu.org/software/grub/manual/html_node/BIOS-installation.html#BIOS-installation
+image.partition(new='1:4MiB:+1MiB')
```
Comment (zyga) [inline examples/proto.py]: I think we want to separate the index, the offset and the size in the constructor. The particular `sgdisk` directive is easily computed from the internal representation. Please disregard this comment if this already happens below (reading top-to-bottom).

File: examples/proto.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+image.partition(typecode='1:21686148-6449-6E6F-744E-656564454649')
```
Comment (zyga) [inline examples/proto.py]: Same note about the typecode, I'd split the partition number from the GUID type.

File: examples/proto.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+image.partition(change_name='1:grub')
```
Comment (zyga) [inline examples/proto.py]: And again, from the name

**Comment by zyga** [comment]
I did a quick review. We should land this and iterate. What I'd like to do next (when time permits) is to model an `image.yaml` file, allow to validate the model and render it to a script or executed it using this code above.

Please disregard my first comment. As I said, it was a top-down review and the comment is not useful.

**Comment by warsaw** [comment]
Thanks for the review!  I think the comment is useful, but of course this was a pretty straight translation from shell to Python.  You might indeed be right that it would be better to split those arguments, but I also agree this should land and we can iterate.  We may find that generating it from the `image.yaml` will make the best API obvious.

**Comment by zyga** [comment]
Thank you for landing this.

**Pull request merged**